api/src/wfl/service/datarepo.clj

```diff
@@ -94,16 +93,15 @@
 
 (defn poll-job
   "Poll the job with `job-id` every `seconds` [default: 5] and return its
-   result."
-  ([job-id seconds]
-   (let [result   #(get-repository-json "jobs" job-id "result")
-         running? #(-> (get-repository-json "jobs" job-id)
-                       :job_status
-                       #{"running"})]
-     (while (running?) (.sleep TimeUnit/SECONDS seconds))
-     (result)))
-  ([job-id]
-   (poll-job job-id 5)))
+   result with `max-attempts` [default: 20]."
+  [job-id & [seconds max-attempts]]
+  (let [result   #(get-repository-json "jobs" job-id "result")
+        done? #(-> (get-repository-json "jobs" job-id)
+                   :job_status
+                   #{"running"}
```

Comment:
Are there intermediate states other than "running"?

Reply:
No - It's running, succeeded or failed. I'm checking that it's not running, just `poll` returns when the value is non-nil.

Reply:
I'm unsure what's better, checking if it's running then inverting or checking if it's either succeeded or failed.

Reply:
My instinct is that it's more likely that intermediate active states are added vs. more final states added.  It might be good to pull these known states out like how we do for [Cromwell](https://github.com/broadinstitute/wfl/blob/develop/api/src/wfl/service/cromwell.clj#L12-L18) and [Rawls](https://github.com/broadinstitute/wfl/blob/develop/api/src/wfl/service/rawls.clj#L12-L25), then check against the final state list.  Will make future debugging easier if the states change upstream.

Reply:
Good idea. I'll do that.